src/libraries/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -303,21 +303,16 @@ public static class ClientOperations
                                 return;
                             }
 
-                            string? name = e.InnerException?.GetType().Name;
-                            switch (name)
+                            if (ctx.HttpVersion == HttpVersion.Version20 &&
+                                e is HttpProtocolException protocolException &&
+                                (protocolException.ErrorCode == 0x2 || // INTERNAL_ERROR, UseKestrel (https://github.com/dotnet/aspnetcore/issues/12256)
+                                 protocolException.ErrorCode == 0x8))  // CANCEL, UseHttpSys
                             {
-                                case "Http2ProtocolException":
-                                case "Http2ConnectionException":
-                                case "Http2StreamException":
-                                    if ((e.InnerException?.Message?.Contains("INTERNAL_ERROR") ?? false) || // UseKestrel (https://github.com/dotnet/aspnetcore/issues/12256)
-                                        (e.InnerException?.Message?.Contains("CANCEL") ?? false)) // UseHttpSys
-                                    {
-                                        return;
-                                    }
-                                    break;
+                                return;
                             }
                         }
 
+                        // TODO: change when HttpProtocolException gets implemented for HTTP/3
```

Comment:
@rkzikm shall we address this:
1. As part of #71432
2. In this PR after merging #71432
3. In a completely separate PR
?